nemo/collections/nlp/data/machine_translation/preproc_mt_data.py

```diff
@@ -944,7 +958,54 @@ def write_parallel_batches_to_tarfiles(
             decoder_bpe_dropout=decoder_bpe_dropout,
             decoder_model_name=decoder_model_name,
             decoder_r2l=decoder_tokenizer_r2l,
+            encoder_tokenizer_legacy=encoder_tokenizer_legacy,
+            decoder_tokenizer_legacy=decoder_tokenizer_legacy,
         )
+
+        # validate no token is negative for sentencepiece tokenizers and add missing special tokens.
+        for tok_name, tok_library, tok_model, legacy in [
```

Comment:
I would want to avoid putting legacy support in places like this script as later removal could be difficult if things rely on this parameter being there. 
We should probably just have the legacy support in the megatron NMT model and the tarred dataset creation script for immediate use.

Reply:
Same as the above point. If you want to pre-process a dataset with the mT5 tokenizer, your tarred data will silently contain `-1` everywhere you insert <bos> and you may only catch the error while training, where the model will fail with a not-so-helpful CUDA index assert check.